woocommerce/woocommerce-android
# Magic-link sent screen experiment

This PR integrates the changes in the Login library (wordpress-mobile/WordPress-Login-Flow-Android#88) and uses a Remote config parameter (which will be used by an A/B test) to show either the old "magic-link sent" screen or the improved one.

~~**Note:** Please wait before the [Login library PR](wordpress-mobile/WordPress-Login-Flow-Android/pull/88) and the [Survey A/B test PR](https://github.com/woocommerce/woocommerce-android/pull/7074) before merging this PR.~~

**To test:**
1. Start the app and proceed with logging in with a site address
2. Enter the store address and then email address
3. Tap on `Get a login link by email` button
4. Notice the original "magic-link sent" screen is displayed:

![image](https://user-images.githubusercontent.com/1522856/182183474-4836cff8-e2bc-4d2e-b544-8bcc99bb7097.png)

5. Now comment out the contents of the `FirebaseRemoteConfigRepository::fetchRemoteConfig()` and change the `MagicLinkSentScreenVariant.CONTROL.name` to `MagicLinkSentScreenVariant.IMPROVED.name` of `defaultValues` in the same class
6. Build and run the app again
7. Repeat the same login steps as before
8. After requesting the magic link, notice the improved screen is now displayed:

![image](https://user-images.githubusercontent.com/1522856/183367181-81355a48-9404-4d60-b648-b63461fc88cf.png)
